src/main/java/org/mastodon/mamut/classification/ClassifyLineagesController.java

```diff
@@ -291,6 +419,73 @@ public void setVisualisationParams( final boolean showDendrogram )
 		this.showDendrogram = showDendrogram;
 	}
 
+	public void setExternalProjects( final File[] projects, final boolean addTagSetToExternalProjects )
```

Comment:
I would recommend remove the ProjectSession, MastodonProjectService classes. They don't bring any advantage for the code. Instead I would recommend to add a class ExternalProjects which has this setExternalProjects(File[]) method, a close() method and probably some getter methods....